services/repository/generate.go

```diff
@@ -12,7 +12,12 @@ import (
 
 // GenerateRepository generates a repository from a template
 func GenerateRepository(doer, owner *models.User, templateRepo *models.Repository, opts models.GenerateRepoOptions) (*models.Repository, error) {
-	generateRepo, err := models.GenerateRepository(doer, owner, templateRepo, opts)
+	ctx, sess, err := models.TxDBContext()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
May be a `defer sess.Close()`? And then an actual `sess.Close()` before calling `notification.NotifyCreateRepository()` because that function will also open a session.

It's a little bit messy, however. If you want to make it tighter because you are bored, I think you could do:

```
err := models.WithTx( func(ctx DBContext) error {
   ... your code ...
})
```

That should open and close the session for you, but I've never attempted that. 😄

Reply:
Done.